### ISSUE TITLE
Could list characters be  ...not characters?

### ISSUE THREAD

**Issue by Skarfrael** [giuspen/cherrytree#935]
Hello all! One of my greatest torments in using this great application is the list characters. I mean the numbers for numbered lists and circles, squares etc. for unnumbered ones. I face 2 main problems.

The **first** is that they are set in the text as characters.

As @txe mentioned at another issue (that also frustrates me about lists), here:
"...for now bullets/numbers/todo lists is just text [...]" _Originally posted by @txe in https://github.com/giuspen/cherrytree/issues/849#issuecomment-634274423_

I would really like for these characters to not mess with normal text, in selection, deletion etc.

By contrast, when previewing Markdown in other apps, list characters aren't selected when the text in a list is selected, and that's the behavior I would like, since bullets are also not copied, pasted etc. when the rest of the text is.

For example, the below selection in cherrytree would include the circles:
![image](https://user-images.githubusercontent.com/40795688/85038510-543dd700-b18f-11ea-840f-4194a5f1c9b4.png)

Also, when deleting the space on the left of a bullet point I would like to remove 1 indentation from this list item, not just delete a space. But I guess that's relevant on using Tabs instead of Spaces? But the "Insert spaces instead of tabs" preference doesn't seem to have any effect on it.

A **second** issue is moving them. When selecting 2 or more list items and hitting tab/shift-tab to move them left and right, the "type" doesn't change. I have to manually hit tab for each list item separately, which is a bit tedious.

Here is an example, of just selecting and using tab:
![image](https://user-images.githubusercontent.com/40795688/85039619-97e51080-b190-11ea-8492-fd1bcdc5df18.png)

For example, at the third picture all item point should have the circles of the 1st level.

In any case, I would love it for these 2 issues to be solved since I love the app. 
Keep on the great work!

**Comment by ForeverRainbow** [comment]
I was thinking of opening an issue about this actually. I did wonder why cherrytree uses unicode bullets for lists, it is actually a very easy way of doing it, but as you have highlighted there are some problems. 

I was thinking maybe all lists (todo, bulleted, ordered, etc) could become widgets, not sure how viable that is @txe but if they could then indentation and changes would be much more consistant and require less guesswork afaik. Also if TOC's could be implemented as a widget then they could potentially be dynamic and update themselves.

**Comment by txe** [comment]
Using widgets in such role maybe is not that good, they may look really unnatural to users, e.g. when selecting text with widgets. I looked through many gtk apps, and they all use widgets to show pictures or emeji. I'd like to know if there is a gtk editor with rich text support like in Word.

@Skarfrael, about your first issue, I'm pretty sure that text was rendered on some surface with Cairo lib (if it was gtk) like in pdf documents. So text looks like the text, and spec symbols (dots) is just images. It's quite easy to do, but this approach is good for them because they don't need to edit text in place or have editor features on such surface. In our case it's not possible, we use an editor control which operates symbols and styles like in html. 
If you know any gtk application which has such features, there is a great chance that we can do it too. But I don't think so.

About the second issue, it is doable to fix.

**Comment by Skarfrael** [comment]
> So text looks like the text, and spec symbols (dots) is just images. It's quite easy to do, but this approach is good for them because they don't need to edit text in place or have editor features on such surface

Yeah, that's why I mentioned it is Markdown preview. The screenshot is from Joplin btw.

> If you know any gtk application which has such features, there is a great chance that we can do it too. But I don't think so.

Sadly I don't have much relevant experience to make any proposals.


Thanks for looking into it anyway.

**Comment by metal450** [comment]
I agree wholeheartedly with both of these issues, though in my case the 1st is much more impactful (as copying & pasting to/from lists is a more frequent activity than altering their indentation levels).

In #849, it was proposed that the issue of paragraph indentation might be solved by Scintilla.  I wonder if this might somehow apply to the 1st issue as well - such that lists could be handled as "indented paragraphs" with the list indicators outside of those paragraphs, or something to that effect?
